### PR TITLE
feature: adds facial recognition models to vectorizer feature

### PR DESCRIPTION
**Pull request by bsbodden**
Add a FACIAL/FACE detection and vectorization specific models to the `@Vectorize` annotation.

### CODE REVIEW

File: redis-om-spring/src/main/java/com/redis/om/spring/vectorize/FeatureExtractor.java [hunk 1]
```diff
@@ -96,6 +110,15 @@ public void processEntity(byte[] redisKey, Object item, boolean isNew) {
                 logger.warn("Error generating image embedding", e);
               }
             }
+            case FACE -> {
+              Resource resource = applicationContext.getResource(fieldValue.toString());
+              try {
+                byte[] feature = getFacialImageEmbeddingsFor(resource.getInputStream());
+                accessor.setPropertyValue(vectorize.destination(), feature);
+              } catch (IOException e) {
```
Comment (sazzad16) [inline redis-om-spring/src/main/java/com/redis/om/spring/vectorize/FeatureExtractor.java:118]: ^

File: redis-om-spring/src/test/java/com/redis/om/spring/annotations/vectorize/face/FaceDetectionTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  @Test
+  void testFaceEmbedding() throws TranslateException, IOException {
+    Resource resource = applicationContext.getResource("classpath:/images/morgan.jpg");
+    Image img = ImageFactory.getInstance().fromInputStream(resource.getInputStream());
+    try (Predictor<Image, float[]> predictor = faceEmbeddingModel.newPredictor()) {
+      float[] embedding = predictor.predict(img);
+      byte[] embeddingAsByteArray = floatArrayToByteArray(embedding);
+      System.out.println(embeddingAsByteArray.length);
```
Comment (sazzad16) [inline redis-om-spring/src/test/java/com/redis/om/spring/annotations/vectorize/face/FaceDetectionTest.java:70]:
```suggestion
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: redis-om-spring/src/main/java/com/redis/om/spring/vectorize/FeatureExtractor.java [hunk 3]
```diff
@@ -73,6 +77,16 @@ public byte[] getImageEmbeddingsFor(InputStream is) {
     }
   }
 
+  public byte[] getFacialImageEmbeddingsFor(InputStream is) {
+    try (Predictor<Image, float[]> predictor = faceEmbeddingModel.newPredictor()) {
+      var img = imageFactory.fromInputStream(is);
+      return ObjectUtils.floatArrayToByteArray(predictor.predict(img));
+    } catch (IOException | TranslateException e) {
```
[the remark this answers is not in the document]
Comment (bsbodden) [reply redis-om-spring/src/main/java/com/redis/om/spring/vectorize/FeatureExtractor.java]: Good catch!

File: redis-om-spring/src/test/java/com/redis/om/spring/annotations/vectorize/face/FaceDetectionTest.java (hunk 2 above)
Comment (bsbodden) [reply redis-om-spring/src/test/java/com/redis/om/spring/annotations/vectorize/face/FaceDetectionTest.java]: Cleaned!

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**